src/init/detectProject.ts

```diff
@@ -0,0 +1,107 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { findFiles } from '../utils/vscode-extensions';
+import { IVSIcons } from './../models/contributions';
+import {
+  showCustomizationMessage,
+  reload,
+  applyCustomization,
+  togglePreset,
+  cancel,
+} from '../commands';
+import { extensionSettings } from '../settings';
+import { messages as msg } from '../messages';
+import { getConfig } from '../utils/vscode-extensions';
+
+export function detectProject(vsicons: IVSIcons): PromiseLike<boolean> {
+  if (vsicons.projectDetection.disableDetect) {
+    return Promise.resolve(false) as PromiseLike<boolean>;
+  }
+
+  return findFiles('package.json', '**/node_modules/**', 1)
+    .then((res) => {
+      if (!res || !res.length) {
+        return false;
+      }
+
+      const project = fs.readFileSync(res[0].fsPath, "utf8");
+      const projectJson = (typeof project === "string") ? JSON.parse(project) : null;
+
+      if (!projectJson) {
+        return false;
+      }
+
+      return checkForAngular2Project(projectJson, vsicons.presets.angular) || false;
```

Comment:
Just FYI, in case other project detection support is added in the future, this `return` statement can be extended.
i.e.
```
return checkForAngular2Project(projectJson, vsicons.presets.angular)
 || checkForEmberProject(projectJson, vsicons.presets.ember)
 || checkForMeteorProject(projectJson, vsicons.presets.meteor)
 || false;
``